shithead_game_server/src/lobby/mod.rs

```diff
@@ -75,59 +123,82 @@ impl Lobby {
         let (broadcast_messages_sender, _) = broadcast::channel(BROADCAST_CHANNEL_CAPACITY);
         Self {
             state: LobbyState::Waiting,
-            deck: CardsDeck::shuffled(),
-            trash: CardsDeck::empty(),
-            current_turn: None,
-            owner_id,
-            id,
-            name,
-            player_list: players,
-            broadcast_messages_sender,
+            data: LobbyNonStateData {
+                owner_id,
+                id,
+                name,
+                player_list: players,
+                broadcast_messages_sender,
+            },
         }
     }
 
+    /// Returns a reference to the lobby's broadcast messages sender.
+    pub fn broadcast_messages_sender(&self) -> &broadcast::Sender<ServerMessage> {
+        &self.data.broadcast_messages_sender
+    }
+
+    /// The id of the lobby
+    pub fn id(&self) -> LobbyId {
+        self.data.id
+    }
+
     /// The amount of players in the lobby.
     pub fn players_amount(&self) -> usize {
-        self.player_list.len()
+        self.data.player_list.len()
     }
 
     /// Is the lobby empty?
     pub fn is_empty(&self) -> bool {
-        self.player_list.is_empty()
+        self.data.player_list.is_empty()
     }
 
     /// The current state of the lobby.
-    pub fn state(&self) -> LobbyState {
-        self.state
+    pub fn state(&self) -> &LobbyState {
+        &self.state
+    }
+
+    /// Is the lobby in a waiting state?
+    pub fn is_waiting(&self) -> bool {
+        matches!(&self.state, LobbyState::Waiting)
     }
 
     /// The name of the lobby.
     pub fn name(&self) -> &str {
-        &self.name
+        &self.data.name
     }
 
     /// The id of the owner.
     pub fn owner_id(&self) -> ClientId {
-        self.owner_id
+        self.data.owner_id
     }
 
     /// The ids of the players in the lobby.
     pub fn player_ids<'a>(&'a self) -> impl Iterator<Item = ClientId> + 'a {
-        self.player_list.player_ids()
+        self.data.player_list.player_ids()
     }
 
     /// Returns a list of exposed information about each player.
     pub fn exposed_player_list(&self) -> Vec<ExposedLobbyPlayerInfo> {
         self.player_ids().map(ExposedLobbyPlayerInfo::new).collect()
     }
 
+    /// Returns an in game lobby context for this lobby, if it is in an in game state.
+    fn in_game_context(&mut self) -> Option<InGameLobbyContext> {
+        match &mut self.state {
+            LobbyState::GameStarted(state) => Some(InGameLobbyContext::new(state, &mut self.data)),
+            _ => None,
+        }
+    }
```

Comment:
Why would you create an `InGameLobbyContext` every time you need it, instead of implementing all it's methods inside the `InGameLobbyState`?

Reply:
`InGameLobbyState` is missing some information about the lobby that is required to perform these operations.